### PR TITLE
Temporarily revert use of Iterable.castFrom().

### PR DESCRIPTION
**Pull request by jakobr-google**
Internal repo has not been rolled to latest Dart SDK yet.

### CODE REVIEW

**Comment by Hixie** [comment]
please remove the code rather than commenting it out (and probably the TODO as well)

cc @mraleph

File: packages/flutter/lib/src/widgets/nested_scroll_view.dart [hunk 1]
```diff
@@ -890,8 +890,9 @@ class _NestedScrollController extends ScrollController {
   }
 
   Iterable<_NestedScrollPosition> get nestedPositions sync* {
+    yield* positions;
```
Comment (mraleph) [inline packages/flutter/lib/src/widgets/nested_scroll_view.dart]: Please do this:

```dart
for (ScrollPosition pos in positions) {
  _NestedScrollPosition nestedScrollPos = pos;
  yield nestedScrollPos;
}
```

instead and leave  // TODO

**Review by mraleph — CHANGES_REQUESTED** [review]

**Comment by Hixie** [comment]
New plan. Looks like `castFrom` has now landed in Google's internal version of dart. So I think this can be closed.

**Comment by jakobr-google** [comment]
So it seems, yes! I'd checked earlier today, but we should be good now. Never mind this, then!

**Pull request closed without merging**